src/Perlang.ConsoleApp/PerlangConsole.cs

```diff
@@ -0,0 +1,117 @@
+#nullable enable
+using System;
+using System.CommandLine.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using Perlang.Exceptions;
+using Perlang.Stdlib;
+using String = Perlang.Lang.String;
+
+namespace Perlang.ConsoleApp;
+
+// Shamelessly inspired by System.CommandLine.IO.SystemConsole,
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+public class PerlangConsole : IPerlangConsole
+{
+    public PerlangConsole()
+    {
+        Error = StandardErrorStreamWriter.Instance;
+        Out = StandardOutStreamWriter.Instance;
+    }
+
+    /// <inheritdoc />
+    public IStandardStreamWriter Error { get; }
+
+    /// <inheritdoc />
+    // TODO: Can we do this without relying on System.Console?
+    public bool IsErrorRedirected => Console.IsErrorRedirected;
+
+    /// <inheritdoc />
+    public IStandardStreamWriter Out { get; }
+
+    /// <inheritdoc />
+    // TODO: Can we do this without relying on System.Console?
+    public bool IsOutputRedirected => Console.IsOutputRedirected;
+
+    /// <inheritdoc />
+    public bool IsInputRedirected => Console.IsInputRedirected;
+
+    public void WriteStdoutLine(String s)
+    {
+        int bytesWritten = Libc.Internal.write(Libc.Internal.STDOUT_FILENO, s.Bytes, (int)s.Length);
```

Comment:
Interestingly enough, this approach seems to work on both Linux and windows (tested on Windows 10). :astonished: If we don't need to write platform-dependent code, we _should_ obviously not be writing platform-dependent code...

(Reading keypresses might be a bigger challenge. We'll get to that eventually, trying to do without `Console.ReadKey`)